Content.Server/_CorvaxGoob/Deathmatch-CS/CSRuleSystem.cs

```diff
@@ -0,0 +1,165 @@
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<GameRunLevelChangedEvent>(MapClearing);
+        SubscribeLocalEvent<GameRuleStartedEvent>(RoundStart);
+        SubscribeLocalEvent<RoundRestartCleanupEvent>(RoundEnd);
+
+        SubscribeLocalEvent<FighterComponent, MobStateChangedEvent>(OnKillReported);
+        SubscribeLocalEvent<FighterComponent, PlayerDetachedEvent>(PlayerHasDisconnectednected);
+        SubscribeLocalEvent<FighterComponent, EraseEvent>(EraseАPlayer);
+        SubscribeLocalEvent<FighterComponent, EntityTerminatingEvent>(DeleteАPlayer);
+    }
+
+    private void RoundStart(ref GameRuleStartedEvent _)
+    {
+        CreateNewSession();
+    }
+    private void RoundEnd(RoundRestartCleanupEvent _)
+    {
+        _sessions.Clear();
+    }
+
+    public void CreateNewSession()
+    {
+        var query = EntityQueryEnumerator<CSRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uId, out var csRuleC, out var gRuleC))
+        {
+            if (!GameTicker.IsGameRuleActive(uId, gRuleC))
+                return;
+
+            if ((this._sessions?.Count ?? 0) < csRuleC.NumberOfSessions)
+            {
+                Session newSession = new();
+                GameMapPrototype? protoMap;
+
+                if (!csRuleC.RandomArena)
+                    protoMap = _gameMapManager.GetSelectedMap();
+                else
+                {
+                    var maps = _gameMapManager.CurrentlyEligibleMaps().ToList();
+                    protoMap = _random.Pick(maps);
+                }
+
+                Addmap(out newSession.MapId, protoMap);
+                _sessions?.Add(newSession);
+            }
+        }
+    }
+    private void Addmap(out MapId mapId, GameMapPrototype? mapProto)
+    {
+        if (mapProto == null)
+        {
+            _gameMapManager.SelectMapByConfigRules();
+            mapProto = _gameMapManager.GetSelectedMap();
+        }
+        GameTicker.LoadGameMap(mapProto!, out MapId mapIdproxy);
+        mapId = mapIdproxy;
+        _map.InitializeMap(mapId);
+    }
+
+    private void MapClearing(GameRunLevelChangedEvent ev)
+    {
+        var query = EntityQueryEnumerator<CSRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uId, out _, out var gRuleC))
+        {
+            if (!GameTicker.IsGameRuleActive(uId, gRuleC))
+                return;
+
+            var activeMapIds = new HashSet<MapId>(_sessions.Select(s => s.MapId));
+            foreach (var mapId in _map.GetAllMapIds())
+            {
+                if (_map.MapExists(mapId) && !activeMapIds.Contains(mapId))
+                {
+                    _map.DeleteMap(mapId);
+                }
+            }
+        }
+    }
+
+    private void OnKillReported(EntityUid uid, FighterComponent isFighterComp, MobStateChangedEvent args)
+    {
+        if (MobState.Dead != args.NewMobState || _sessions == null)
```

Comment:
```suggestion
        if (_sessions == null)
            return;
        if (MobState.Dead != args.NewMobState)
            return;
```
Их тут вполне можно разделить.